demos/tests/cases.py

```diff
@@ -25,71 +25,70 @@
 MONITORS = {'-u': 'cdm'}
 TestCase = collections.namedtuple('TestCase', ['options'])
 
+
 class Demo:
+    def __init__(self, subdirectory, implementation, device_keys=None, test_cases=None):
+        self.subdirectory = subdirectory + '/' + implementation if implementation else subdirectory
```

Comment:
```suggestion
        self.subdirectory = subdirectory + '/' + implementation
```

All demos have implementations, so the conditional isn't needed.

Reply:
Done. I have a little doubt in case of setting empty implemenation value, but we would solve it later.